compiler/rustc_span/src/lib.rs

```diff
@@ -795,6 +795,24 @@ impl Span {
         })
     }
 
+    /// Splits a span into two composite spans around a certain position.
+    ///
+    /// ```text
+    ///     --------------------
+    ///     self lorem ipsum end
+    ///     <<<<<<<<<<<<<<>>>>>>
+    /// ```
+    pub fn split(self, pos: u32) -> (Span, Span) {
```

Comment:
conventionally, this is named `split_at`: https://doc.rust-lang.org/nightly/std/index.html?search=split_at